src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletSupport.scala

```diff
@@ -0,0 +1,268 @@
+      oldPubKeys.foreach(storage.addKey)
+      storage.removePaths()
+    }
+  }
+
+  protected def processUnlock(state: ErgoWalletState, masterKey: ExtendedSecretKey, usePreEip3Derivation: Boolean)(implicit addrEncoder: ErgoAddressEncoder): Try[ErgoWalletState] = {
+    log.info("Starting wallet unlock")
+    convertLegacyClientPaths(state.storage, masterKey)
+    // Now we read previously stored, or just stored during the conversion procedure above, public keys
+    // If no public keys in the database yet, add master's public key into it
+    if (state.storage.readAllKeys().toIndexedSeq.isEmpty) {
+      if (usePreEip3Derivation) {
+        // If usePreEip3Derivation flag is set in the wallet settings, the first key is the master key
+        val masterPubKey = masterKey.publicKey
+        state.storage.addKey(masterPubKey)
+        log.info("Wallet unlock finished using usePreEip3Derivation")
+        Try(updatePublicKeys(state, masterKey, Vector(masterPubKey)))
+      } else {
+        // If no usePreEip3Derivation flag is set, add first derived key (for m/44'/429'/0'/0/0) to the db
+        deriveNextKeyForMasterKey(state, masterKey, usePreEip3Derivation).flatMap { case (derivationResult, newState) =>
+          derivationResult.result.map { case (_, _, firstSk) =>
+            val firstPk = firstSk.publicKey
+            newState.storage.addKey(firstPk)
+            newState.storage.updateChangeAddress(P2PKAddress(firstPk.key))
+            log.info("Wallet unlock finished")
+            updatePublicKeys(newState, masterKey, Vector(firstPk))
+          }
+        }
+      }
+    } else {
+      log.info("Wallet unlock finished using existing keys in storage")
+      Success(state)
+    }
+  }
+
+
+  /**
+    * Convert requests (to make payments or to issue an asset) to transaction outputs
+    * There can be only one asset issuance request in the input sequence.
+    *
+    * @param requests - an input sequence of requests
+    * @return sequence of transaction outputs or failure if inputs are incorrect
+    */
+  protected def requestsToBoxCandidates(
+                                         requests: Seq[TransactionGenerationRequest],
+                                         assetId: BoxId,
+                                         fullHeight: Int,
+                                         parameters: Parameters,
+                                         publicKeyAddresses: Seq[P2PKAddress]): Try[Seq[ErgoBoxCandidate]] =
+    Traverse[List].sequence {
+      requests.toList
+        .map {
+          case PaymentRequest(address, value, assets, registers) =>
+            Success(new ErgoBoxCandidate(value, address.script, fullHeight, assets.toColl, registers))
+          case AssetIssueRequest(addressOpt, valueOpt, amount, name, description, decimals, registers) =>
+            // Check that auxiliary registers do not try to rewrite registers R0...R6
+            if (registers.exists(_.exists(_._1.number < 7))) {
```

Comment:
I changed this from `if (registers.exists(_.forall(_._1.number < 7)))` as I assume we don't want any 0-7 register to be overriden. Is that correct or ?

Reply:
Oh I guess you're right. Would be good to cover this with a test.

Reply:
I covered that already, I was just not sure if the AssetIssueRequest aux registers really cannot override any of the R0-R6 https://github.com/ergoplatform/ergo/pull/1311/files#diff-fca9ab9b22f435644bbc31e76ac9b4e4fb37e61555dd05e51a1c3763ad8f9ff1R90-R93

Reply:
I covered that already, I was just not sure if the AssetIssueRequest aux registers really cannot override any of the R0-R6 https://github.com/ergoplatform/ergo/pull/1311/files#diff-fca9ab9b22f435644bbc31e76ac9b4e4fb37e61555dd05e51a1c3763ad8f9ff1R90-R93

Reply:
Yes, that was an intention (aux registers shouldn't override R0-R6)